astropy/timeseries/tests/test_downsample.py

```diff
@@ -314,6 +345,26 @@ def test_downsample_edge_cases(time, time_bin_start, time_bin_end):
         )  # single-valued time series falls in *first* bin
 
 
+@pytest.mark.parametrize("masked_cls", [MaskedColumn, Masked(u.Quantity)])
+def test_downsample_with_masked_array(masked_cls):
+    # See gh-17991
+    m = masked_cls([0.0, 1.0, -np.inf, 3.0, 4.0], mask=False)
+    m.mask[2] = True
+    ts = TimeSeries(time=INPUT_TIME, data=[m], names=["m"])
+    assert isinstance(ts["m"], masked_cls)
+    down = aggregate_downsample(
+        ts, time_bin_start=INPUT_TIME[:-1:2], time_bin_end=INPUT_TIME[1::2]
+    )
+    expected = masked_cls([0.5, 3.0], mask=False)
+    assert_array_equal(down["m"], expected)
+    m.mask[3] = True
+    down2 = aggregate_downsample(
+        ts, time_bin_start=INPUT_TIME[:-1:2], time_bin_end=INPUT_TIME[1::2]
+    )
+    expected2 = masked_cls([0.5, 3.0], mask=[False, True])
```

Comment:
See https://github.com/astropy/astropy/pull/18023#issuecomment-2816361077  (hard to write inline)